src/cascadia/QueryExtension/ExtensionPalette.xaml

```diff
@@ -0,0 +1,384 @@
+<!--
+    Copyright (c) Microsoft Corporation. All rights reserved. Licensed under
+    the MIT License. See LICENSE in the project root for license information.
+-->
+<UserControl x:Class="Microsoft.Terminal.Query.Extension.ExtensionPalette"
+             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
+             xmlns:local="using:Microsoft.Terminal.Query.Extension"
+             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+             xmlns:mux="using:Microsoft.UI.Xaml.Controls"
+             VerticalAlignment="Stretch"
+             AllowFocusOnInteraction="True"
+             AutomationProperties.Name="{x:Bind ControlName, Mode=OneWay}"
+             IsTabStop="True"
+             LostFocus="_lostFocusHandler"
+             PointerPressed="_rootPointerPressed"
+             PreviewKeyDown="_previewKeyDownHandler"
+             TabNavigation="Cycle"
+             mc:Ignorable="d">
+
+    <UserControl.Resources>
+        <ResourceDictionary>
+            <mux:StackLayout x:Name="VerticalStackLayout"
+                             Orientation="Vertical"
+                             Spacing="16" />
+            <DataTemplate x:Key="QueryMessageTemplate"
+                          x:DataType="local:ChatMessage">
+                <Grid Height="Auto"
+                      Margin="4"
+                      HorizontalAlignment="Right">
+                    <Grid.RowDefinitions>
+                        <RowDefinition Height="Auto" />
+                        <RowDefinition Height="Auto" />
+                    </Grid.RowDefinitions>
+                    <StackPanel Grid.Row="1"
+                                MaxWidth="400"
+                                Padding="16,8,16,8"
+                                CornerRadius="8">
+                        <StackPanel.Background>
+                            <SolidColorBrush Color="#005EB7" />

```

Comment:
In the future, we need to use existing xaml color ramp items. Otherwise, we lose theme support.